internal/client/client.go

```diff
@@ -15,22 +15,26 @@ type AuthStrategyType string
 var AuthStrategy = struct {
 	SERVICE_TOKEN              AuthStrategyType
 	UNIVERSAL_MACHINE_IDENTITY AuthStrategyType
+	OIDC_MACHINE_IDENTITY      AuthStrategyType
 }{
 	SERVICE_TOKEN:              "SERVICE_TOKEN",
 	UNIVERSAL_MACHINE_IDENTITY: "UNIVERSAL_MACHINE_IDENTITY",
+	OIDC_MACHINE_IDENTITY:      "OIDC_MACHINE_IDENTITY",
 }
 
 type Config struct {
 	HostURL string
 
-	AuthStrategy AuthStrategyType
+	AuthStrategy          AuthStrategyType
+	IsMachineIdentityAuth bool
 
 	// Service Token Auth
 	ServiceToken string
 
 	// Universal Machine Identity Auth
 	ClientId     string
 	ClientSecret string
```

Comment:
Nit: Ideally I think we should keep the credentials as an interface on the config, so we avoid having a cluttered config. We do something along the lines of this in the Infisical Agent.

I can take a deeper look into this once we add more auth methods though!